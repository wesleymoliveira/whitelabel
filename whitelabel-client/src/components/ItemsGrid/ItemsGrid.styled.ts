import { Box, styled } from '@mui/material';

export const Wrapper = styled(Box)`
	width: 100%;
	display: grid;
	grid-template-columns: repeat(1fr, 350px);
	justify-content: center;
`;
