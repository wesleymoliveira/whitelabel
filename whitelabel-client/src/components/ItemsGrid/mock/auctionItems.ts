export const auctionItems = {
	auction_items: [
		{
			artist: 'test1',
			courtesy_of: 'test1',
			description: 'test1',
			id: 1,
			image:
				'https://res.cloudinary.com/dgop0ic8b/image/upload/v1650298086/items/loveisintheair_qo5uzc.png',
			is_for_auction: true,
			lot: 10,
			name: 'test1',
			previous_bid: '$100.00',
			price: '$100.00',
			slug: 'test1',
			video: 'test1',
			highest_bid: '1500'
		}
	]
};
