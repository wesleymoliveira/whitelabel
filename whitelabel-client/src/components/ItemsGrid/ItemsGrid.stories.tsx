import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { ItemsGrid } from './ItemsGrid';
import { auctionItems } from './mock/auctionItems';

export default {
	title: 'components/ItemsGrid',
	component: ItemsGrid
} as Meta;

export const Default: Story = () => <ItemsGrid {...auctionItems} />;
