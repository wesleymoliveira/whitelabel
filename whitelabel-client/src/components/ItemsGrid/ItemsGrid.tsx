import { AuctionItems } from '@/typings/shared';
import { Box, Typography } from '@mui/material';
import Image from 'next/image';
import { Link } from '@/components/Link';
import * as S from './ItemsGrid.styled';
import { formatMoney } from '@/utils/price';

export function ItemsGrid({ auction_items }: AuctionItems) {
	return (
		<S.Wrapper data-testid='items-grid'>
			{auction_items?.map(({ id, image, name, slug, highest_bid, price }) => (
				<Box key={id}>
					<Box
						sx={{
							position: 'relative',
							width: 350,
							height: 350,
							cursor: 'pointer',
							'& img': {
								border: '1px solid #000000',
								borderRadius: '30px'
							},
							pt: '10px'
						}}
					>
						<Link href={`/item/${slug}`}>
							<Image src={image} width='350' height='350' layout='fixed' alt={name} />
						</Link>
					</Box>
					<Box
						sx={{
							display: 'flex',
							flexDirection: 'column',
							alignItems: 'center',
							my: 3
						}}
					>
						<Typography
							fontSize='14px'
							fontWeight={500}
							fontFamily='Outfit'
							lineHeight='18px'
							letterSpacing='0.38px'
						>
							{highest_bid ? 'Current Top bid' : 'Starting Price'}
						</Typography>
						<Typography
							fontSize='30px'
							fontWeight={400}
							fontFamily='Outfit'
							lineHeight='41px'
							letterSpacing='0.38px'
							pb='4px'
						>
							{formatMoney(highest_bid ? highest_bid : price)}
						</Typography>
						<Image
							src='https://res.cloudinary.com/dgop0ic8b/image/upload/v1650298104/icons/geo_rounded_voaq5i.svg'
							width='103'
							height='23'
							alt='Art'
						/>
					</Box>
				</Box>
			))}
		</S.Wrapper>
	);
}
