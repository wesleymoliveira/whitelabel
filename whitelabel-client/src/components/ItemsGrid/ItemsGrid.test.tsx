import { render, screen } from '@testing-library/react';

import { ItemsGrid } from './ItemsGrid';
import { auctionItems } from './mock/auctionItems';

describe('<ItemsGrid />', () => {
	it('should render the component', () => {
		const { container } = render(<ItemsGrid {...auctionItems} />);

		expect(screen.getByTestId(/items-grid/i)).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});
});
