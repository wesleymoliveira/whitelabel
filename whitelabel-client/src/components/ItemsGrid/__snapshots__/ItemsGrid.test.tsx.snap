// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsGrid /> should render the component 1`] = `
.emotion-0 {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(1fr, 350px);
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 24px;
  margin-bottom: 24px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  font-size: 14px;
  font-weight: 500;
  font-family: Outfit;
  line-height: 18px;
  letter-spacing: 0.38px;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  font-size: 30px;
  font-weight: 400;
  font-family: Outfit;
  line-height: 41px;
  letter-spacing: 0.38px;
  padding-bottom: 4px;
}

<div
  class="MuiBox-root emotion-0"
  data-testid="items-grid"
>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
  >
    <span
      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27350%27%20height=%27350%27/%3e"
        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
      />
    </span>
    <img
      alt="test1"
      data-nimg="intrinsic"
      decoding="async"
      href="/item/test1"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
  <div
    class="MuiBox-root emotion-1"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-2"
    >
      Current Top bid
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-3"
    >
      1500
    </p>
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27103%27%20height=%2723%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Art"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
</div>
`;
